partner_preview_scripts/encore_parser.py

```diff
@@ -0,0 +1,410 @@
+from pyspark.sql.types import StringType, StructType, StructField, ArrayType, FloatType
+from pyspark.sql.dataframe import DataFrame
+
+from common.evidence import initialize_sparksession, write_evidence_strings, GenerateDiseaseCellLines
+
+class EncoreEvidenceGenerator:
+    def __init__(self, spark: SparkSession, cell_passport_df: DataFrame, shared_parameters: dict) -> None:
+        self.spark = spark
+        self.cell_passport_df = cell_passport_df
+
+        # Parsing paramters:
+        self.log_fold_change_cutoff_p_val = shared_parameters["logFoldChangeCutoffPVal"]
+        self.log_fold_change_cutoff_FDR = shared_parameters["logFoldChangeCutoffFDR"]
+        self.interaction_cutoff_p_val = shared_parameters["interactionCutoffPVal"]
+        self.data_folder = shared_parameters["data_folder"]
+        self.release_version = shared_parameters["releaseVersion"]
+        self.release_date = shared_parameters["releaseDate"]
+
+    @staticmethod
+    @udf(ArrayType(StructType([
+        StructField("targetFromSourceId", StringType(), False),
+        StructField("targetRole", StringType(), False),
+        StructField("interactingTargetFromSourceId", StringType(), False),
+        StructField("interactingTargetRole", StringType(), False),
+    ])))
+    def parse_targets(gene_pair: str, gene_role: str) -> dict:
+        """The gene pair string is split and assigned to the relevant role + exploding into evidence of two targets.
+
+        gene pair: 'SHC1~ADAD1', where 'SHC1' is the library gene, and 'ADAD1' is the anchor gene.
+
+        Both genes will be targetFromSource AND interactingTargetFromSource, while keeping their roles.
+        """
+        genes = gene_pair.split('~')
+        roles = [
+            gene_role.replace('Combinations', '').lower(),
+            'anchor'
+        ]
+
+        assert(len(genes) == 2)
+        parsed = []
+
+        for i, (gene, role) in enumerate(zip(genes, roles)):
+            parsed.append({
+                'targetFromSourceId': gene,
+                'targetRole': role,
+                'interactingTargetFromSourceId': genes[1] if i == 0 else genes[0],
+                'interactingTargetRole': roles[1] if i == 0 else roles[0]
+            })
+
+        return parsed
+
+    def get_lfc_data(self, lfc_file: str) -> DataFrame:
+        """Reading log fold change file and generating a dataframe."""
+
+        # Fixed statistical field names:
+        stats_fields = ['p-value', 'fdr', 'lfc']
+
+        # Reading the data into a single dataframe:
+        lfc_df = self.spark.read.csv(lfc_file, sep=' ', header=True)
+
+        # Collect the cell lines from the lfc file header:
+        cell_lines = set(['_'.join(x.split('_')[:-1]) for x in lfc_df.columns[4:]])
+
+        # Generating struct for each cell lines:
+        # SIDM00049_CSID1053_p-value
+        # SIDM00049_CSID1053_fdr
+        # SIDM00049_CSID1053_lfc
+        # Into: SIDM00049_CSID1053: struct(p-value, fdr, lfc)
```

Comment:
And also, again sorry if it's a stupid question, but what exactly are those values? I can see that the LFC dataset contains these values for several gene pairs in several model cell lines, but what exactly is being measured, and compared to what?

(This is the type of information I think would be extremely helpful to include in README, or even better just into the header of this module)

Reply:
I also added explanation.